pkg/types/upgrades.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+type ChangeReport struct {
+	ComponentReports []ComponentChangeReport
+}
+
+type ComponentChangeReport struct {
+	ComponentName string
+	OldVersion    string
```

Comment:
How do we plan on using the old version? Can't we just have a hash map instead of this struct and only add to hashmap with component as key and new version as value if there is a change and check based on that?

Reply:
Interesting. What's the benefit of using a map? Imo it's way more complex to expand it. And we don't need to do lookups anywhere.

My idea was that every "owner" of components would have a method to return these. That way we can check what needs to change before actually changing it. I think we'll use it for validations/pre-flight checks.
Also, not in scope for this feature, but it could be used to have a `upgrade cluster plan` command